edb/server/compiler/sertypes.py

```diff
@@ -396,6 +396,8 @@ def _describe_type(self, t, view_shapes, view_shapes_metadata,
                 buf.append(el_name_bytes)
                 buf.append(_uint16_packer(self.uuid_to_pos[el_type]))
 
+            if self.inline_typenames:
+                self._add_annotation(mt, type_id.bytes)
```

Comment:
Is type_id a name or an id here?

Reply:
It's a UUID of the derived object type, we need this to link the annotation back to the original object type descriptor.

Reply:
In what cases do you need this?

I don't like to abuse the field that is called "name" to store IDs in it when it's convenient for us.

Reply:
I mean we should always just inline plain object names, not IDs. If we need IDs then we need to reconsider how we implement this change (e.g. add a new annotation type).

Reply:
Oh I think you misunderstood. As a reminder, the data structure of type descriptors is like the following:

```
uint8 type = 1     # ObjectShapeDescriptor
uuid id = 111...   # Type ID of the object
...                # Data of the 111 ObjectShapeDescriptor

uint8 type = 3     # ScalarTypeDescriptor
uuid id = 222...   # Type ID of the user-defined scalar type
uint16 base_idx =  # Index of the typedesc for the base type

uint8 type = 0xff  # TypeAnnotationDescriptor
uuid id = 111...   # References 111 ObjectShapeDescriptor
string name = ...  # Full type name of 111

uint8 type = 0xff  # TypeAnnotationDescriptor
uuid id = 222...   # References 222 ScalarTypeDescriptor
string name = ...  # Full type name of 222
```

For `ScalarTypeDescriptor`, the `material_type()` is itself, so we could just pass in the `mt` to `_add_annotation()`; but for `ObjectShapeDescriptor`, the current type is derived from the`mt = material_type()`, thus their IDs are different, we still need to pass in the `type_id` of the current type so that the annotation could reference the correct object typedesc.

Reply:
Oh my, I'm sorry Fantix. Can we rename `_add_annotation` to `_add_typename_annotation`?

Reply:
No problem!